clippy_lints/src/doc/too_long_first_doc_paragraph.rs

```diff
@@ -0,0 +1,92 @@
+use rustc_ast::ast::Attribute;
+use rustc_errors::Applicability;
+use rustc_hir::{Item, ItemKind};
+use rustc_lint::LateContext;
+
+use clippy_utils::diagnostics::{span_lint, span_lint_and_then};
+use clippy_utils::source::snippet_opt;
+
+use super::TOO_LONG_FIRST_DOC_PARAGRAPH;
+
+pub(super) fn check(
+    cx: &LateContext<'_>,
+    item: &Item<'_>,
+    attrs: &[Attribute],
+    mut first_paragraph_len: usize,
+    check_private_items: bool,
+) {
+    if !check_private_items && !cx.effective_visibilities.is_exported(item.owner_id.def_id) {
+        return; // We don't check private items in this case.
+    }
+    if first_paragraph_len <= 200
+        || !matches!(
+            item.kind,
+            ItemKind::Static(..)
+                | ItemKind::Const(..)
+                | ItemKind::Fn(..)
+                | ItemKind::Macro(..)
+                | ItemKind::Mod(..)
+                | ItemKind::TyAlias(..)
+                | ItemKind::Enum(..)
+                | ItemKind::Struct(..)
+                | ItemKind::Union(..)
+                | ItemKind::Trait(..)
+                | ItemKind::TraitAlias(..)
```

Comment:
I feel like we can extract this to `mod.rs`. While it shouldn't change performance, it makes more sense for a "can have documentation" check to exist there.

Reply:
So the list isn't exactly the one matching "items that can have documentation" but instead is only "items that can have documentation and are on the module page". I'll add a comment to mention it.